src/textual/_animator.py

```diff
@@ -374,6 +374,24 @@ def _animate(
         self._idle_event.clear()
         self._complete_event.clear()
 
+    def stop_animation(self, obj: object, attribute: str) -> None:
```

Comment:
If we are stopping an animation, it could be left in a state before its natural end point. I think we should have an option to set the animate to its final destination.

I think we also have `on_complete` callbacks when an animation is complete. It doesn't appear that these will be called with this change.

Reply:
Good point. The "value never reaches conclusion" thing felt like a natural consequence of stopping early (so yeah, I guess making it an option makes sense when doing the stop). Calling any `on_complete` is something that did pass me by.

Just to be clear: do we want `on_complete` to be called no matter, or only if there's a "stop and complete"? Or should we perhaps have a `on_stopped` to handle both cases?

Reply:
I think just `on_complete` for now. Until we have a request / use case for a `on_stopped`.

Reply:
Noted. And on the matter of if it should always be called?

I'm thinking I'll add a `complete=True|False` to `stop_animation` (likely defaulting to `True`); so would we expect it to be called *only* if `on_complete` is `True`?